src/main/kotlin/com/terraformation/backend/daily/JobRunrRecoveryThread.kt

```diff
@@ -0,0 +1,79 @@
+package com.terraformation.backend.daily
+
+import com.terraformation.backend.config.TerrawareServerConfig
+import com.terraformation.backend.log.perClassLogger
+import java.sql.SQLException
+import java.time.Duration
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import javax.annotation.ManagedBean
+import javax.annotation.PostConstruct
+import javax.annotation.PreDestroy
+import javax.sql.DataSource
+import org.jobrunr.server.BackgroundJobServer
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+
+/**
+ * Restarts JobRunr's background job processing thread after database problems. JobRunr shuts down
+ * when it can't talk to the database to avoid spewing errors to the log, but it doesn't come back
+ * up when the database becomes available again. We want the server to recover from transient
+ * database problems.
+ */
+@ConditionalOnProperty(TerrawareServerConfig.DAILY_TASKS_ENABLED_PROPERTY, matchIfMissing = true)
```

Comment:
This is so we don't start this background thread in contexts where we're not running JobRunr at all, e.g., when you run `./gradlew generateOpenApiDocs` and the server starts up with no database configuration.